test/test_path_sim.cpp

```diff
@@ -0,0 +1,156 @@
+#include "PathSimulator.hpp"
+#include "algorithms/Entanglement.hpp"
+#include "algorithms/Grover.hpp"
+#include "dd/Export.hpp"
+
+#include <gtest/gtest.h>
+#include <memory>
+
+using namespace dd::literals;
+
+TEST(TaskBasedSimTest, SimpleCircuitSingleThreaded) {
```

Comment:
At the moment, everything is single threaded anyway. So we can just drop the `SingleThreaded` from all the test names and just introduce `MultiThreaded` once this is possible.